### DISCUSSION TITLE
桌面歌词样式改善、歌词操作改善、本地歌词查询改善，等

### DISCUSSION THREAD

**Discussion by TaicEart** [zhongyang219/MusicPlayer2#439] [General]
### 能否自定义桌面歌词的样式？
目前只能是阴影，且向右下偏移，且为纯黑，这样就有几个问题：

1. 如果字色也设为黑，看起来像重影
2. 如果字色设为白，在许多程序和网页背景上面只能看到字的影子，比如白天的github
3. 在较花的背景下难以看清

改善不必重复造轮子，可直接参考 Windows（win10以上）设置→轻松使用→隐藏式字幕，或B站的cc字幕功能。
- [ ] 待解…

### ~能否限制双行桌面歌词的左右尺寸？~
~桌面歌词设为双行时上下歌词飞到屏幕左右了，相隔甚远，难以阅读。~ 
- [x] 解决。

### 能否自定义歌词空行时显示替代？
指「Music…」和「♪♪♪」可以自定义。
- [ ] 待解…

### 能否自定义滚轮操作？
我觉得 https://github.com/zhongyang219/MusicPlayer2/issues/248  “将主窗口界面的滚轮功能更改为滚动歌词，并可通过歌词定位播放” 可以考虑，但这个滚轮操作应当限定在歌词和进度条区域，界面上其它地方还是调音量。某些播放器有类似功能。
- [ ] 待解…

### 能否支持多个本地歌词目录？
我觉得 https://github.com/zhongyang219/MusicPlayer2/issues/413  “能不能像foobar的eslyric插件一样支持多个默认的歌词搜索目录？”是一个需求，同为本地播放器的千千静听也支持类似的功能。
- [ ] 待解…

顺带，我将9页开启的issues都翻了一遍，其中有些人需求与我上述的类似：

1.  https://github.com/zhongyang219/MusicPlayer2/issues/119 
2.  https://github.com/zhongyang219/MusicPlayer2/issues/357
3.  https://github.com/zhongyang219/MusicPlayer2/issues/62

**Comment by zhongyang219** [comment]
桌面歌词的尺寸可以是可以设置的，将桌面歌词解锁后，点击“歌词背景穿透”按钮，去掉“歌词背景穿透状态”，看到桌面歌词出现白色的半透明背景时，就可以调整桌面歌词窗口的大小了。
![image](https://user-images.githubusercontent.com/30562462/205905667-7dbb5ab7-b6ab-4c95-9230-6d94c57e1534.png)

**Reply by TaicEart** [reply]
这个解决了👌
